vassal-app/src/main/java/VASSAL/build/module/Console.java

```diff
@@ -159,31 +163,74 @@ else if (matches("?", option) || matches("help", option) || ("".equals(option)))
 
 
   private boolean doProperty() {
-    final String option   = nextString("");
+    final String first = nextString("");
+    final String option;
+    final boolean useSelected;
+    if (first.toLowerCase().startsWith("sel")) { //NON-NLS
+      option = nextString("");
+      useSelected = true;
+    }
+    else {
+      option = first;
+      useSelected = false;
+    }
     final String property = nextString("");
 
     if (matches("?", option) || matches("help", option)) { //NON-NLS
       show("Usage:"); //NON-NLS
-      show("  /property show [property]        - show global property value"); //NON-NLS
-      show("  /property set [property] [value] - set global property new value"); //NON-NLS
+      show("  /property [selected] show [property]        - show global property value"); //NON-NLS
+      show("  /property [selected] set [property] [value] - set global property new value"); //NON-NLS
+      show("  If optional keyword 'selected' included, show or set is for property of selected piece(s). "); //NON-NLS
+      show("  NOTE: many piece trait properties cannot be set (e.g. 'ObscuredToOthers', 'Invisible'), and the attempt to set them will simply fail quietly. "); //NON-NLS
     }
-    else if (matches("show", option)) { //NON-NLS
-      final MutableProperty.Impl propValue = (MutableProperty.Impl) GameModule.getGameModule().getMutableProperty(property);
-      if (propValue != null) {
-        show("[" + property + "]: " + propValue.getPropertyValue());
+    else if (matches("show", option) || matches("set", option)) { //NON-NLS
+
+      final GameModule gm = GameModule.getGameModule();
+
+      if (useSelected) {
+        final List<GamePiece> selected = KeyBuffer.getBuffer().asList();
+
+        if ((selected != null) && !selected.isEmpty()) {
```

Comment:
You don't need to check if it's non-empty here---the loop just won't loop over an empty list.

Reply:
Oh, there's an `else` down there. Never mind!

Reply:
Wait, no, that `else` is further out in scope.

Reply:
fixed